lib/publiccloud/ec2.pm

```diff
@@ -123,12 +124,14 @@ sub upload_img {
           . "-d 'OpenQA tests' "
           . ($sec_group  ? "--security-group-ids '" . $sec_group . "' " : '')
           . ($vpc_subnet ? "--vpc-subnet-id '" . $vpc_subnet . "' "     : '')
+          . ($ami_id     ? "--ec2-ami '" . $ami_id . "' "               : '')
           . "'$file'",
         timeout => 60 * 60
     );
 
     my $ami = $self->find_img($img_name);
     die("Cannot find image after upload!") unless $ami;
```

Comment:
in case `$ami_id` is specified it should be equal to what we getting from `find_img` , right ? are we ok with scenario when `$ami_id != $ami` ?

Reply:
The ami, which is specified here, is used to build the image. It is an existing one. I usually use
`pint amazon images --active --region eu-central-1` to find a valid one.
This ami can be specified inside the `ec2utils.conf` file. But it was shown that it is usefull to specify one. Also if we now are moving to use different regions.

Reply:
So `$ami_id != $ami` is always true.

Reply:
Comment added, to explain the situation.